proto/frontend/src/device_mgr.cpp

```diff
@@ -475,10 +475,17 @@ class DeviceMgrImp {
   Status direct_meter_write(p4::Update_Type update,
                             const p4::DirectMeterEntry &meter_entry,
                             const SessionTemp &session) {
-    if (!check_p4_id(meter_entry.meter_id(), P4ResourceType::METER))
-      return make_invalid_p4_id_status();
-
+    if (!meter_entry.has_table_entry()) {
+      RETURN_ERROR_STATUS(Code::UNIMPLEMENTED,
+                          "Modifying ALL direct meters is not supported yet");
+    }
     const auto &table_entry = meter_entry.table_entry();
+    if (table_entry.table_id() == 0) {
+      RETURN_ERROR_STATUS(Code::UNIMPLEMENTED,
+                          "Modifying ALL direct meters is not supported yet");
+    }
```

Comment:
what about the case where `table_entry.match_size() == 0`

Reply:
Not allowing wildcard matching for now. entry_handle_from_table_entry(table_entry, &entry_handle); will return an INVALID_ARGUMENT error.